src/ControlSystem/ControlErrors/Translation.hpp

```diff
@@ -0,0 +1,138 @@
+ * \vec{v}) \f$, \f$ \mathbf{q} \f$ is the quaternion from the \link
+ * domain::CoordinateMaps::TimeDependent::Rotation Rotation \endlink map, \f$ a
+ * \f$ is the function \f$ a(t) \f$ from the \link
+ * domain::CoordinateMaps::TimeDependent::CubicScale CubicScale \endlink map,
+ * \f$ \mathbf{\delta q}\wedge\mathbf{c}_B \equiv (0, \delta\vec{q} \times
+ * \vec{c}_B) \f$, \f$ \delta\vec{q} \f$ is the \link
+ * control_system::ControlErrors::Rotation Rotation \endlink control error, and
+ * \f$ \delta a\f$ is the \link control_system::ControlErrors::Expansion
+ * Expansion \endlink control error.
+ *
+ * Requirements:
+ * - This control error requires that there be exactly two objects in the
+ *   simulation
+ * - Currently both these objects must be black holes
+ * - Currently this control system can only be used with the \link
+ *   control_system::Systems::Translation Translation \endlink control system
+ * - There must exist an expansion map and a quaternion rotation map in the
+ *   coordinate map with names "Expansion" and "Rotation", respectively.
+ */
+struct Translation : tt::ConformsTo<protocols::ControlError> {
+  using options = tmpl::list<>;
+  static constexpr Options::String help{
+      "Computes the control error for translation control. This should not "
+      "take any options."};
+
+  void pup(PUP::er& /*p*/) {}
+
+  template <typename Metavariables, typename... TupleTags>
+  DataVector operator()(const Parallel::GlobalCache<Metavariables>& cache,
+                        const double time,
+                        const std::string& /*function_of_time_name*/,
+                        const tuples::TaggedTuple<TupleTags...>& measurements) {
+    const auto& domain = get<domain::Tags::Domain<3>>(cache);
+    const auto& functions_of_time = get<domain::Tags::FunctionsOfTime>(cache);
+
+    using quat = boost::math::quaternion<double>;
+
+    const quat quaternion = datavector_to_quaternion(
+        functions_of_time.at("Rotation")->func(time)[0]);
+    const double expansion_factor =
+        functions_of_time.at("Expansion")->func(time)[0][0];
+
+    using center_B = control_system::QueueTags::Center<::ah::ObjectLabel::B>;
+
+    const DataVector grid_position_of_B = array_to_datavector(
+        domain.excision_spheres().at("ObjectBExcisionSphere").center());
+    const DataVector& current_position_of_B = get<center_B>(measurements);
+
+    const DataVector rotation_error =
+        rotation_control_error_(cache, time, "Rotation", measurements);
+    // Use B because it's on the positive x-axis, however, A would work as well.
+    // Just so long as we are consistent.
+    const DataVector rotation_error_cross_grid_pos_B =
+        cross(rotation_error, grid_position_of_B);
+
+    const double expansion_error =
+        expansion_control_error_(cache, time, "Expansion", measurements)[0];
+
+    // From eq. 42 in 1304.3067
+    const quat middle_expression = datavector_to_quaternion(
```

Comment:
eq. 42 has an overall factor of expansion_factor out in front, that appears to be missing here.  Is that intentional?

Reply:
The overall factor of expansion_factor is applied a few lines below when I do the full calculation.

Reply:
Ah, ok, I missed that.